packages/perspective-viewer-d3fc/src/js/legend/legend.js

```diff
@@ -6,119 +6,53 @@
  * the Apache License 2.0.  The full license can be found in the LICENSE file.
  *
  */
-import * as d3Legend from "d3-svg-legend";
+
+import scrollableLegend from "./scrollableLegend";
 import {getChartElement} from "../plugin/root";
-import {getOrCreateElement} from "../utils/utils";
-import legendControlsTemplate from "../../html/legend-controls.html";
 
+const scrollLegend = scrollableLegend();
 export function legend(container, settings, colour) {
     if (colour) {
-        const groupSize = 15;
-        const numberOfGroups = Math.ceil(colour.domain().length / groupSize);
-        const doesLegendScroll = numberOfGroups > 1;
-
-        if (doesLegendScroll) {
-            const reloadLegend = index => createLegend(container, settings, colour, doesLegendScroll, groupSize, index);
-            legendScrolling(container, settings, numberOfGroups, reloadLegend);
-        }
-        createLegend(container, settings, colour, doesLegendScroll, groupSize, settings.legendGroupIndex);
-    }
-}
-
-function legendScrolling(container, settings, numberOfGroups, reloadLegend) {
-    if (!settings.legendGroupIndex) settings.legendGroupIndex = 0;
-    const legendControls = getLegendControls(container);
-
-    const setPageText = () => legendControls.select("#page-text").text(`${settings.legendGroupIndex + 1}/${numberOfGroups}`);
-    setPageText();
-    const upArrow = legendControls.select("#up-arrow");
-    const downArrow = legendControls.select("#down-arrow");
+        scrollLegend
+            .scale(colour)
+            .shape("circle")
+            .shapeRadius(6)
+            .orient("vertical")
+            .on("cellclick", function(d) {
+                settings.hideKeys = settings.hideKeys || [];
+                if (settings.hideKeys.includes(d)) {
+                    settings.hideKeys = settings.hideKeys.filter(k => k !== d);
+                } else {
+                    settings.hideKeys.push(d);
+                }
+
+                getChartElement(this).draw();
+            });
 
-    upArrow.on("click", function() {
-        if (settings.legendGroupIndex > 0) {
-            settings.legendGroupIndex--;
-            legendPage(downArrow);
+        if (settings.mainValues.length <= 1) {
+            scrollLegend.labels(options => {
+                const parts = options.domain[options.i].split("|");
+                return parts.slice(0, parts.length - 1).join("|");
+            });
         }
-        deactivateArrow(upArrow, 0);
-    });
-    deactivateArrow(upArrow, 0);
 
-    downArrow.on("click", function() {
-        if (settings.legendGroupIndex < numberOfGroups - 1) {
-            settings.legendGroupIndex++;
-            legendPage(upArrow);
+        let legendSelection = container.select("div.legend-container");
+        if (legendSelection.size() === 0) {
+            legendSelection = container.append("div");

```

Comment:
Should this use the new get or create element function?

Reply:
I actually just reverted `legend.js` to what it was like before (except for using `scrollableLegend` instead of `d3Legend`). If you want to take over again and put `getOrCreateElement` in here, then that seems fine to me.